main/main.c

```diff
@@ -147,12 +147,11 @@ static void guiTask(void *pvParameter) {
     create_demo_application();
 
     while (1) {
-        /* Delay 1 tick (assumes FreeRTOS tick is 10ms */
-        vTaskDelay(pdMS_TO_TICKS(10));
-
         /* Try to take the semaphore, call lvgl related function on success */
         if (pdTRUE == xSemaphoreTake(xGuiSemaphore, portMAX_DELAY)) {
-            lv_task_handler();
+            uint32_t ticks_to_wait = lv_task_handler();
+            if(ticks_to_wait == 0) continue;
```

Comment:
Should be considered as '''unwanted''' due to prevent mutex release.

Sorry

Reply:
I discovered also that using function return with lower than default HZ rate in FreeRTOS (which is 1000) makes display very "laggy". Which makes sense but maybe some1 will be using this like this and experience that symptom.

Reply:
> I discovered also that using function return with lower than default HZ rate in FreeRTOS (which is 1000) makes display very "laggy". Which makes sense but maybe some1 will be using this like this and experience that symptom.

What did you mean by this? I think I am having this issue of lag.